### PR TITLE
Update to new httpx api

### LINKED ISSUE #90: Rename httpx.AsyncClient --> httpx.Client
To follow upstream:
https://github.com/encode/httpx/pull/544/commits/7898bff8f6fa26ff26f4484a139f027181336a26

### PR DESCRIPTION
**Pull request by dhirschfeld**
Rename AsyncClient --> Client

Fixes #90

### CODE REVIEW

File: gidgethub/httpx.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
 import asyncio
 from typing import Mapping, Tuple, Any
 
-import httpx
+try:
+    from httpx import AsyncClient as Client
+except ImportError:
```
Comment (brettcannon) [inline gidgethub/httpx.py]: We only support starting at Python 3.6, so...

```suggestion
except ModuleNotFoundError:
```

File: gidgethub/test/test_httpx.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import datetime
 
-import httpx
+try:
+    from httpx import AsyncClient as Client
+except ImportError:
```
Comment (brettcannon) [inline gidgethub/test/test_httpx.py]:
```suggestion
except ModuleNotFoundError:
```

**Comment by dhirschfeld** [comment]
Then change `s/ImportError/ModuleNotFoundError/` has resulted in ImportErrors so I'll go ahead and revert that
```
==================================== ERRORS ====================================
________________ ERROR collecting gidgethub/test/test_httpx.py _________________
ImportError while importing test module '/home/travis/build/brettcannon/gidgethub/gidgethub/test/test_httpx.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
gidgethub/test/test_httpx.py:4: in <module>
    from httpx import AsyncClient as Client
E   ImportError: cannot import name 'AsyncClient'
```

**Comment by brettcannon** [comment]
@dhirschfeld mind updating the changelog (and making sure to thank yourself)?

File: docs/changelog.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 Changelog
 =========
 
+vNext
+'''''
+
+- Adapt to the new ``httpx`` api to support versions >= 0.8.0
```
Comment (brettcannon) [inline docs/changelog.rst]:
```suggestion
- Adapt to the new ``httpx`` api to support versions >= 0.8.0 (thanks [Dave Hirschfeld](https://github.com/dhirschfeld))
```

File: gidgethub/httpx.py [hunk 4]
```diff
@@ -1,13 +1,16 @@
 import asyncio
 from typing import Mapping, Tuple, Any
 
-import httpx
+try:
+    from httpx import AsyncClient as Client  # type: ignore
+except ImportError:
+    from httpx import Client
```
Comment (brettcannon) [inline gidgethub/httpx.py]: I assume the old sync class was (unfortunately) named `Client`?

Comment (dhirschfeld) [reply gidgethub/httpx.py]: I think so. IIUC there is no longer a sync client - httpx is async only (for now). There may be plans afoot to reintroduce a sync client but I'm not fully across all of that.

With httpx & hip, async http libraries are a rapidly evolving field in Python!

**Comment by brettcannon** [comment]
Thanks!

**Comment by brettcannon** [comment]
@dhirschfeld I can't make any promises on a timeline, but I will try to get a release out when I can.

**Comment by brettcannon** [comment]
Looks like httpx is currently broken: https://travis-ci.org/brettcannon/gidgethub/jobs/632071215. Seems `Response` changed and no longer has a `read()` method: https://www.encode.io/httpx/api/#response.

**Comment by brettcannon** [comment]
Opened #92 to track the breakage.

**Pull request merged**